apps/mull-ui/src/utilities.ts

```diff
@@ -16,3 +22,46 @@ export const formatDate = (
 
   return { year: date.getFullYear(), month, day: date.getDate(), time: timeString };
 };
+
+/**
+ * Utility method for logging in the user. Used in the login page and registration page
+ *
+ * @param login the login mutation
+ * @param loginInput the login input containing the variables for the mutation
+ * @param updateToast
+ * @param setAccessToken
+ * @param setUserId
+ * @param history
+ *
+ * @see RegisterPage
+ * @see LoginPage
+ */
+export const loginUser = async (
+  login,
+  loginInput: ILoginForm,
+  updateToast: (type: TypeOptions, message: string) => void,
+  setAccessToken: React.Dispatch<React.SetStateAction<string>>,
+  setUserId: React.Dispatch<React.SetStateAction<number>>,
+  history: History
+) => {
+  try {
+    var { data, errors } = await login({ variables: { loginInput } });
+  } catch (err) {
+    updateToast(toast.TYPE.ERROR, err.message);
+    return;
+  }
+
+  if (errors) {
+    const messages = errors.map((err) => err.message).reduce((err1, err2) => err1 + err2);
+    updateToast(toast.TYPE.ERROR, `Error during login: ${messages}`);
+    return;
+  }
+
+  const accessToken = data.login.accessToken;
+  setAccessToken(accessToken);
+
+  const decodedToken = jwtDecode(accessToken) as { id: number };
```

Comment:
You should check if the `decodedToken` is valid. Since a falsy or malformed token will throw an InvalidTokenError error.

Reply:
Oh good point, I completely forgot about that, will do